src/basic/locale-util.c

```diff
@@ -270,6 +272,94 @@ bool is_locale_utf8(void) {
         return (bool) cached_answer;
 }
 
+static Set *keymaps = NULL;
+
+static int nftw_cb(
+                const char *fpath,
+                const struct stat *sb,
+                int tflag,
+                struct FTW *ftwbuf) {
+
+        char *p, *e;
+        int r;
+
+        if (tflag != FTW_F)
+                return 0;
+
+        if (!endswith(fpath, ".map") &&
+            !endswith(fpath, ".map.gz"))
+                return 0;
+
+        p = strdup(basename(fpath));
+        if (!p)
+                return log_oom();
```

Comment:
also, in library code, please do not log directly, at anything higher than LOG_DEBUG. Library code should propagate errors, so that the caller logs, and not log itself. See CODING_STYLE for details about this.